File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/pref/SettingsFragment.java
package com.andruid.magic.newsdaily.pref;

import android.os.Bundle;

import androidx.fragment.app.DialogFragment;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

import com.andruid.magic.newsdaily.R;
import com.andruid.magic.newsdaily.util.PrefUtil;

import java.util.Objects;

public class SettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.app_preferences);
        CountryPreference countryPreference = findPreference(getString(R.string.pref_country));
        String defCountry = PrefUtil.getDefaultCountry(Objects.requireNonNull(getContext()));
        String country = PreferenceManager.getDefaultSharedPreferences(getContext()).getString(
                getString(R.string.pref_country), defCountry);
        if (countryPreference != null)
            countryPreference.setCountry(country);
    }

    @Override
    public void onDisplayPreferenceDialog(Preference preference) {
        DialogFragment dialogFragment = null;
        if(preference instanceof CountryPreference)
            dialogFragment = CountryPreferenceDialogFragment.newInstance(preference.getKey());
        if(dialogFragment != null){
            dialogFragment.setTargetFragment(this, 0);
            dialogFragment.show(getParentFragmentManager(), "countryPref");
        }
        else
            super.onDisplayPreferenceDialog(preference);
    }
}

File: NewsDailyJava/app/src/main/java/com/andruid/magic/newsdaily/pref/SettingsFragment.kt
package com.andruid.magic.newsdaily.pref

import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceManager
import com.andruid.magic.newsdaily.R
import com.andruid.magic.newsdaily.pref.CountryPreferenceDialogFragment.Companion.newInstance
import com.andruid.magic.newsdaily.util.PrefUtil.Companion.getDefaultCountry

class SettingsFragment : PreferenceFragmentCompat() {

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.app_preferences)
        val countryPreference: CountryPreference? = findPreference(getString(R.string.pref_country))
        val defCountry = getDefaultCountry(context)
        val country = PreferenceManager.getDefaultSharedPreferences(context)
                .getString(getString(R.string.pref_country), defCountry)
        countryPreference?.country = country ?: "in"
    }

    override fun onDisplayPreferenceDialog(preference: Preference) {
        val dialogFragment: DialogFragment?
        if (preference is CountryPreference) {
            dialogFragment = newInstance(preference.getKey())
            dialogFragment?.setTargetFragment(this, 0)
            dialogFragment?.show(parentFragmentManager, "countryPref")
        }
        else
            super.onDisplayPreferenceDialog(preference)
    }
}